modules/core/src/main/scala/com/azavea/stac4s/ItemCollection.scala

```diff
@@ -1,42 +1,73 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
 import io.circe.refined._
+import io.circe.syntax._
+import shapeless.LabelledGeneric
+import shapeless.ops.record.Keys
 
 final case class ItemCollection(
     _type: String = "FeatureCollection",
     stacVersion: StacVersion,
     stacExtensions: List[String],
     features: List[StacItem],
-    links: List[StacLink]
+    links: List[StacLink],
+    extensionFields: JsonObject = ().asJsonObject
 )
 
 object ItemCollection {
 
+  private val generic      = LabelledGeneric[ItemCollection]
+  private val keys         = Keys[generic.Repr].apply
+  val itemCollectionFields = keys.toList.flatMap(field => substituteFieldName(field.name)).toSet
+
```

Comment:
Can we put it into a single function to reduce some boilerplate?

Reply:
That's the thing I had trouble with implicits with -- it wasn't obvious to me how to get the implicit evidence required to have `getFieldNames[T]` cooperate. Specifically, the evidence provided in the example blog post:

```scala
implicit def toAttributes[T, Repr <: HList, KeysRepr <: HList](
    implicit gen: LabelledGeneric.Aux[T, Repr],
    keys: Keys.Aux[Repr, KeysRepr],
    traversable: ToTraversable.Aux[KeysRepr, List, Symbol]
  ):
```

was insufficient for `Keys[generic.Repr]` when I tried making it generic.

Reply:
* It is just an optional comment; 

:+1:

Reply:
Opened https://github.com/azavea/stac4s/issues/89